geode-core/src/main/java/org/apache/geode/internal/cache/DiskStoreImpl.java

```diff
@@ -454,13 +457,15 @@ private static int calcCompactionThreshold(int ct) {
     int length = dirs.length;
     this.directories = new DirectoryHolder[length];
     long tempMaxDirSize = 0;
+
     for (int i = 0; i < length; i++) {
       directories[i] =
           new DirectoryHolder(getName() + "_DIR#" + i, factory, dirs[i], dirSizes[i], i);
 
       if (tempMaxDirSize < dirSizes[i]) {
         tempMaxDirSize = dirSizes[i];
       }
+      this.totalDiskStoreSpace += dirSizes[i] * (1024 * 1024);
```

Comment:
I think there is a bug here: By default, the directory size is `Integer.MAX_VALUE` (see `DiskStoreFactory.DEFAULT_DISK_DIR_SIZE`). Therefore, if the operator does not specify a directory size when creating the disk store, the value for `totalDiskStoreSpace` will be incorrect. I was able to reproduce this by modifying the test `givenDiskStoreIsCreated_whenDataIsStored_thenGetDiskUsagePercentageShouldBeIncreased` to not specify a directory size. When I did that, I saw that the returned value for `getDiskUsagePercentage()` was negative.

Also, that test should probably make sure the disk usage percentage is greater than zero because it still passes in the case I mentioned above.

Another interesting question is, what should `getDiskUsagePercentage()` return when the operator does not specify a directory size when creating the disk store? I'm not sure I have a good answer to this.

Reply:
I posted my opinion in the Jira ticket, lets wait for having an agreement.